### PR TITLE
Sap 46 flink changes

### PR DESCRIPTION
**Pull request by jadbalja**
Changes to make everything used by Flink Serializable.  Plus changes based on making deployments to local Flink cluster and to YARN on phx-dev.

### CODE REVIEW

**Comment by jadbalja** [comment]
Hopefully I've left enough TODOs around that, when you're reviewing, you can tell which parts I already know need work, or at least are in a transitional state.

File: pom.xml [hunk 1]
```diff
@@ -215,6 +447,7 @@
         <configuration>
           <source>1.8</source>
           <target>1.8</target>
+          <!-- use the jdt compiler, because javac doesn't do well with generics in lambdas -->
```
Comment (lucjross) [inline pom.xml]: i'm curious, is there an open bug around generics & lambdas in OpenJDK/Oracle?

File: pom.xml [hunk 2]
```diff
@@ -49,22 +48,37 @@
...
+        </exclusion>
+        <exclusion>
+          <groupId>org.xerial.snappy</groupId>
+          <artifactId>snappy-java</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
 
+    <!-- Spring Boot dependencies -->
+    <dependency>
```
[the remark this answers is not in the document]
Comment (jadbalja) [reply pom.xml]: The idea is not to have conflicting dependencies.  It likely doesn't matter in this particular case, but there are dependencies in here that have dependencies on different versions of things like Guava (between all the things that claim Guava as a dependency, we have three or four different versions specified -- this became a problem when I deployed to Flink due to class loader order differences).

File: src/test/java/com/wellsmith/sapien/persistence/PersistenceWorkflowTest.java [hunk 3]
```diff
@@ -59,7 +60,7 @@ public static void setupClass() {
     System.setProperty("org.slf4j.simpleLogger.defaultLogLevel", "info");
   }
 
-  @Before
+//  @Before
```
[the remark this answers is not in the document]
Comment (jadbalja) [reply src/test/java/com/wellsmith/sapien/persistence/PersistenceWorkflowTest.java]: I think I noted in a TODO that all these tests need to be rewritten and moved to Junit 5.

File: src/main/java/com/wellsmith/sapien/persistence/repository/cassandra/CassandraConfig.java [hunk 4]
```diff
@@ -10,113 +12,47 @@
...
-import org.springframework.data.cassandra.core.CassandraAdminTemplate;
 import org.springframework.data.cassandra.core.cql.keyspace.CreateKeyspaceSpecification;
-import org.springframework.data.cassandra.repository.config.EnableCassandraRepositories;
 import org.springframework.data.convert.CustomConversions;
 import org.springframework.lang.NonNull;
 
+// TODO: is this really even necessary any more?  CassandraCrudRepository is creating its own Cluster
+// and session from CassandraProperties.  Will have to move a lot
+// of configuration logic into its initialize method.
+
```
[the remark this answers is not in the document]
Comment (jadbalja) [reply src/main/java/com/wellsmith/sapien/persistence/repository/cassandra/CassandraConfig.java]: Took a different route that emulates Spring Cassandra autoconfiguration but in a way that actually works.

File: src/main/java/com/wellsmith/sapien/persistence/repository/cassandra/AdditionalCassandraProperties.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
+package com.wellsmith.sapien.persistence.repository.cassandra;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Primary;
+
+@Primary
```
[the remark this answers is not in the document]
Comment (jadbalja) [reply src/main/java/com/wellsmith/sapien/persistence/repository/cassandra/AdditionalCassandraProperties.java]: Sure isn't.  This class doesn't exist any more anyway.

File: src/main/java/com/wellsmith/sapien/persistence/kafka/KafkaConfig.java [hunk 6]
```diff
@@ -10,10 +10,12 @@
 import org.apache.kafka.clients.producer.ProducerConfig;
 import org.apache.kafka.common.config.SaslConfigs;
 import org.springframework.boot.autoconfigure.kafka.KafkaProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.context.annotation.Configuration;
 
 @Configuration
+@ConfigurationProperties
```
[the remark this answers is not in the document]
Comment (jadbalja) [reply src/main/java/com/wellsmith/sapien/persistence/kafka/KafkaConfig.java]: Probably not.  I'm thinking I added it when the properties weren't getting read and I couldn't figure out why.

File: pom.xml (hunk 1 above)
Comment (jadbalja) [reply pom.xml]: I haven't looked.  I assume so.

File: src/main/java/com/wellsmith/sapien/persistence/repository/cassandra/CassandraCrudRepository.java [hunk 7]
```diff
@@ -101,6 +132,10 @@ private void writeObject(ObjectOutputStream out) throws IOException {
 
   private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
     cassandraProperties = (CassandraProperties) in.readObject();
-    initializeRepository();
+    try {
+      initializeRepository();
+    } catch(Exception e) {
+      LOG.error("Exception deserializing CassandraCrudRepository: " );
```
Comment (lucjross) [inline src/main/java/com/wellsmith/sapien/persistence/repository/cassandra/CassandraCrudRepository.java]: after the message, provide `e`

File: src/main/java/com/wellsmith/sapien/persistence/repository/cassandra/CassandraCrudRepository.java [hunk 8]
```diff
@@ -25,25 +39,42 @@
...
-  public CassandraCrudRepository(CassandraProperties cassandraProperties) {
+  public CassandraCrudRepository(CassandraProperties cassandraProperties) throws Exception {
     this.cassandraProperties = cassandraProperties;
     initializeRepository();
   }
 
-  private void initializeRepository() {
+  // TODO: this is all cheating.  Fix it to work with the configuration files.
+  private void initializeRepository() throws Exception {
+    CassandraConfig cassandraConfig = new CassandraConfig(cassandraProperties);
```
Comment (lucjross) [inline src/main/java/com/wellsmith/sapien/persistence/repository/cassandra/CassandraCrudRepository.java]: wha?

**Pull request merged**